syntax = "proto3";

package zdb.kvStore;

service KVStoreService {
    rpc get(GetRequest) returns (GetReply) {}
    rpc set(SetRequest) returns (SetReply) {}
    rpc erase(EraseRequest) returns (EraseReply) {}
    rpc size(SizeRequest) returns (SizeReply) {}
}

message GetRequest {
    string key = 1;
}

message GetReply {
    string value = 1;
}

message SetRequest {
    string key = 1;
    string value = 2;
}

message SetReply {
}

message EraseRequest  {
    string key = 1;
}

message EraseReply {
    string value = 1;
}

message SizeRequest {
}

message SizeReply {
    uint64 size = 1;
}
